### PR TITLE
Rewrite RAND_egd

### PR DESCRIPTION
**Pull request by richsalz**
Use stdio and its buffering.
Limit to 255 bytes (could remove that if neceessary).

### CODE REVIEW

File: crypto/rand/rand_egd.c [hunk 1]
```diff
@@ -68,7 +35,9 @@ int RAND_egd_bytes(const char *path, int bytes)
 {
     return (-1);
```
Comment (paulidale) [inline crypto/rand/rand_egd.c]: Parenthesis removal.
Also copyright date at the top.

File: crypto/rand/rand_egd.c [hunk 2]
```diff
@@ -91,152 +60,92 @@ struct sockaddr_un {
...
-        if (!buf)
-            RAND_seed(tempbuf, egdbuf[0]);
     }
+
+    /* Make request, see how many bytes we can get back. */
+    tempbuf[0] = 1;
+    tempbuf[1] = bytes;
+    if (fwrite(tempbuf, sizeof(char), 2, fp) != 2)
+        goto err;
+    if (fread(tempbuf, sizeof(char), 1, fp) != 1 || tempbuf[0] == 0)
```
Comment (paulidale) [inline crypto/rand/rand_egd.c]: Does there need to be `fflush(fp)` here?

Comment (paulidale) [inline crypto/rand/rand_egd.c]: After the write, before the read.  Its return code should be checked too: daemon dies before the flush but after the write.

**Comment by richsalz** [comment]
additional commit pushed

**Comment by richsalz** [comment]
commit 8e58093 on master; thanks!

File: crypto/rand/rand_egd.c [hunk 3]
```diff
@@ -61,14 +28,16 @@ int RAND_query_egd_bytes(const char *path, unsigned char *buf, int bytes)
 
 int RAND_egd(const char *path)
 {
-    return -1;
+    return (-1);
```
Comment (dot-asm) [inline crypto/rand/rand_egd.c]: I don't quite follow. Styling guide suggests no parenthesis... And it was fixed other way countless times...

**Comment by richsalz** [comment]
There were several of them and, oops, I only fixed many.  I suppose I should do a new PR.

**Comment by levitte** [comment]
What, à whole new PR just for fixing that detail?

**Comment by richsalz** [comment]
Yes, whenever I get around to it.  There’s no other way.

**Pull request closed without merging**